Enable testslide for define_mp_tests

fegin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103628

This diff allows the internal test runs to use testslide.

Differential Revision: [D46741900](https://our.internmc.facebook.com/intern/diff/D46741900/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D46741900/)!

fegin committed:
Enable testslide for define_mp_tests

This diff allows the internal test runs to use testslide.

FB:
`tsloop --mode-opt //caffe2/test/distributed/_spmd:test_transformation -- --focus --fail-fast`
See https://www.internalfb.com/intern/wiki/Test-slide/tsloop/ for the reference

Differential Revision: [D46741900](https://our.internmc.facebook.com/intern/diff/D46741900/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D46741900/)!

[ghstack-poisoned]

fegin committed:
Update on "Enable testslide for define_mp_tests"


This diff allows the internal test runs to use testslide.

FB:
`tsloop --mode-opt //caffe2/test/distributed/_spmd:test_transformation -- --focus --fail-fast`
See https://www.internalfb.com/intern/wiki/Test-slide/tsloop/ for the reference

Differential Revision: [D46741900](https://our.internmc.facebook.com/intern/diff/D46741900/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D46741900/)!

[ghstack-poisoned]

fegin committed:
Update on "Enable testslide for define_mp_tests"


This diff allows the internal test runs to use testslide.

FB:
`tsloop --mode-opt //caffe2/test/distributed/_spmd:test_transformation -- --focus --fail-fast`
See https://www.internalfb.com/intern/wiki/Test-slide/tsloop/ for the reference

Differential Revision: [D46741900](https://our.internmc.facebook.com/intern/diff/D46741900/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D46741900/)!

[ghstack-poisoned]

awgu commented on torch/testing/_internal/common_distributed.py:
```diff
@@ -548,10 +548,10 @@ def wrapper(self):
-        super().__init__(method_name)
-        fn = getattr(self, method_name)
-        setattr(self, method_name, self.join_or_run(fn))
+    def __init__(self, methodName: str = "runTest") -> None:
```
Is testslide requiring this `method_name` -> `methodName` change?

fegin replied:
`method_name` is wrong. PyThon's core unittest.TestCase uses `methodName`. The issue happens because testslide uses kwargs while other test frameworks args. So for other test frameworks, the incorrect name is okay

fegin committed:
Update on "Enable testslide for define_mp_tests"


This diff allows the internal test runs to use testslide.

Differential Revision: [D46741900](https://our.internmc.facebook.com/intern/diff/D46741900/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D46741900/)!

[ghstack-poisoned]

fegin committed:
Update on "Enable testslide for define_mp_tests"


This diff allows the internal test runs to use testslide.

Differential Revision: [D46741900](https://our.internmc.facebook.com/intern/diff/D46741900/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D46741900/)!

[ghstack-poisoned]

fegin committed:
Update on "Enable testslide for define_mp_tests"


This diff allows the internal test runs to use testslide.

Differential Revision: [D46741900](https://our.internmc.facebook.com/intern/diff/D46741900/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D46741900/)!

[ghstack-poisoned]

fegin committed:
Update on "Enable testslide for define_mp_tests"


This diff allows the internal test runs to use testslide.

Differential Revision: [D46741900](https://our.internmc.facebook.com/intern/diff/D46741900/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D46741900/)!

[ghstack-poisoned]

fduwjj reviewed (approved):
Added multi-GPU CI tests.

fegin committed:
Update on "Enable testslide for define_mp_tests"


This diff allows the internal test runs to use testslide.

Differential Revision: [D46741900](https://our.internmc.facebook.com/intern/diff/D46741900/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D46741900/)!

[ghstack-poisoned]

fegin committed:
Update on "Enable testslide for define_mp_tests"


This diff allows the internal test runs to use testslide.

Differential Revision: [D46741900](https://our.internmc.facebook.com/intern/diff/D46741900/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D46741900/)!

[ghstack-poisoned]

fegin committed:
Update on "Enable testslide for define_mp_tests"


This diff allows the internal test runs to use testslide.

Differential Revision: [D46741900](https://our.internmc.facebook.com/intern/diff/D46741900/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D46741900/)!

[ghstack-poisoned]

fegin committed:
Update on "Enable testslide for define_mp_tests"


This diff allows the internal test runs to use testslide.

Differential Revision: [D46741900](https://our.internmc.facebook.com/intern/diff/D46741900/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D46741900/)!

[ghstack-poisoned]

fegin committed:
Update on "Enable testslide for define_mp_tests"


This diff allows the internal test runs to use testslide.

Differential Revision: [D46741900](https://our.internmc.facebook.com/intern/diff/D46741900/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D46741900/)!

[ghstack-poisoned]

fegin committed:
Update on "Enable testslide for define_mp_tests"


This diff allows the internal test runs to use testslide.

Differential Revision: [D46741900](https://our.internmc.facebook.com/intern/diff/D46741900/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D46741900/)!

[ghstack-poisoned]

fegin committed:
Update on "Enable testslide for define_mp_tests"


This diff allows the internal test runs to use testslide.

Differential Revision: [D46741900](https://our.internmc.facebook.com/intern/diff/D46741900/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D46741900/)!

[ghstack-poisoned]

fegin committed:
Update on "Enable testslide for define_mp_tests"


This diff allows the internal test runs to use testslide.

Differential Revision: [D46741900](https://our.internmc.facebook.com/intern/diff/D46741900/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D46741900/)!

[ghstack-poisoned]
